specification/keyvault/data-plane/Microsoft.KeyVault/preview/7.4-preview.1/common.json

```diff
@@ -69,6 +69,59 @@
       },
       "description": "The key vault server error.",
       "x-nullable": true
+    },
+    "LifetimeActions": {
+      "properties": {
+        "trigger": {
+          "$ref": "#/definitions/LifetimeActionsTrigger",
+          "description": "The condition that will execute the action."
+        },
+        "action": {
+          "$ref": "#/definitions/LifetimeActionsType",
+          "description": "The action that will be executed."
+        }
+      },
+      "description": "Action and its trigger that will be performed by Key Vault over the lifetime of a key or secret."
+    },
+    "LifetimeActionsTrigger": {
+      "properties": {
+        "timeAfterCreate": {
+          "type": "string",
+          "description": "Time after creation to attempt to rotate. It only applies to rotate. It will be in ISO 8601 duration format. Example: 90 days : \"P90D\" "
+        },
+        "timeBeforeExpiry": {
+          "type": "string",
+          "description": "Time before expiry to attempt to rotate or notify. It will be in ISO 8601 duration format. Example: 90 days : \"P90D\""
+        }
+      },
+      "description": "A condition to be satisfied for an action to be executed."
+    },
+    "LifetimeActionsType": {
+      "properties": {
+        "type": {
+          "type": "string",
+          "description": "The type of the action.",
+          "enum": [
+            "rotate",
+            "notify"
+          ],
+          "x-ms-enum": {
+            "name": "ActionType",
+            "modelAsString": false,
```

Comment:
If this will be in a response model (which it should), this should be `true`. Doesn't mean you have to support other models, but it will cause deserialization errors if the service returns more enums in the future. See https://github.com/microsoft/api-guidelines/blob/vNext/azure/Guidelines.md#use-extensible-enums. If you are 100% sure this will never change, this should be fine, though.

Reply:
With drawn since this is the same as in 7.3, but please keep in mind for future models.